lib/Serialization/ModuleFile.h

```diff
@@ -344,31 +345,106 @@ class ModuleFile
     return issue;
   }
 
-  /// Emits one last diagnostic, adds the current module details and errors to
-  /// the pretty stack trace, and then aborts.
-  [[noreturn]] void fatal(llvm::Error error) const;
-  void fatalIfNotSuccess(llvm::Error error) const {
-    if (error)
-      fatal(std::move(error));
+  /// This error indicates that the error has already been diagnosed.
+  struct DeserializationError : public llvm::ErrorInfo<DeserializationError> {
```

Comment:
Could you give this type a more precise name and move it with the others in `DeserializationErrors.h`?

If this is just for fatal errors we can keep it in the name, or refer to the usual underlying issue, it's either a format problem or memory corruption for fatal errors.